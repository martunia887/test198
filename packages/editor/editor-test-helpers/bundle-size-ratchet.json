[
  {
    "id": "src.main",
    "name": "main",
    "stats": {
      "size": 41802,
      "gzipSize": 12156
    }
  },
  {
    "id": "node_modules.main",
    "name": "node_modules [main]",
    "stats": {
      "size": 1279065,
      "gzipSize": 362042
    }
  },
  {
    "id": "node_modules.async",
    "name": "node_modules [async]",
    "stats": {
      "size": 1402560,
      "gzipSize": 398050
    }
  },
  {
    "id": "atlaskit.core.main",
    "name": "main",
    "stats": {
      "size": 220187,
      "gzipSize": 48596
    }
  },
  {
    "id": "atlaskit.core.async",
    "name": "async",
    "stats": {
      "size": 169481,
      "gzipSize": 32871
    }
  },
  {
    "id": "atlaskit.editor.main",
    "name": "main",
    "stats": {
      "size": 691319,
      "gzipSize": 183385
    }
  },
  {
    "id": "atlaskit.editor.async",
    "name": "async",
    "stats": {
      "size": 38914,
      "gzipSize": 9079
    }
  },
  {
    "id": "atlaskit.elements.main",
    "name": "main",
    "stats": {
      "size": 81579,
      "gzipSize": 20275
    }
  },
  {
    "id": "atlaskit.elements.async",
    "name": "async",
    "stats": {
      "size": 71999,
      "gzipSize": 21515
    }
  },
  {
    "id": "atlaskit.media.main",
    "name": "main",
    "stats": {
      "size": 185456,
      "gzipSize": 43905
    }
  },
  {
    "id": "atlaskit.media.async",
    "name": "async",
    "stats": {
      "size": 923621,
      "gzipSize": 227183
    }
  },
  {
    "id": "atlaskit.teamwork-ecosystem.main",
    "name": "main",
    "stats": {
      "size": 624,
      "gzipSize": 367
    }
  },
  {
    "id": "combined.main",
    "name": "main",
    "stats": {
      "size": 2489996,
      "gzipSize": 680480
    }
  },
  {
    "id": "combined.async",
    "name": "async",
    "stats": {
      "size": 2605270,
      "gzipSize": 691473
    }
  }
]