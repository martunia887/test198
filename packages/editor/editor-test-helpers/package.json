{
  "name": "@atlaskit/editor-test-helpers",
  "version": "9.11.5",
  "description": "Test Helpers for Atlassian Editor components",
  "license": "Apache-2.0",
  "atlaskit:src": "src/index.ts",
  "module": "dist/esm/index.js",
  "main": "dist/cjs/index.js",
  "types": "dist/cjs/index.d.ts",
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    "Stanislas S"
  ],
  "atlaskit": {
    "internal": true,
    "name": "Editor Test Helpers",
    "team": "editor"
  },
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/adf-schema": "^3.0.0",
    "@atlaskit/analytics-next": "^5.1.2",
    "@atlaskit/editor-common": "^39.17.0",
    "@atlaskit/editor-core": "^112.39.12",
    "@atlaskit/icon": "^19.0.0",
    "@atlaskit/media-core": "^30.0.8",
    "@atlaskit/media-test-helpers": "^24.3.3",
    "@atlaskit/smart-card": "^12.4.1",
    "enzyme": "^3.7.0",
    "prosemirror-commands": "^1.0.7",
    "prosemirror-keymap": "^1.0.1",
    "prosemirror-model": "^1.7.0",
    "prosemirror-state": "^1.2.2",
    "prosemirror-tables": "^0.8.1",
    "prosemirror-utils": "0.9.4",
    "prosemirror-view": "^1.9.12",
    "react-intl": "^2.6.0"
  },
  "peerDependencies": {
    "react": "^16.8.0"
  },
  "devDependencies": {
    "@types/prosemirror-commands": "^1.0.1",
    "@types/prosemirror-keymap": "^1.0.1",
    "@types/prosemirror-model": "^1.5.0",
    "@types/prosemirror-state": "^1.2.0",
    "@types/prosemirror-view": "^1.9.0"
  }
}