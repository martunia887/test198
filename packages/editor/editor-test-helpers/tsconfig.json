{
  "extends": "../../../tsconfig.base.json",
  "compilerOptions": {
    "baseUrl": "./"
  },
  "include": [
    "./src/**/*.ts",
    "./src/**/*.tsx",
    "./tests/**/*.ts",
    "./tests/**/*.tsx",
  ]
}