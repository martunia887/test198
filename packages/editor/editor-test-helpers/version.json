{
	"name": "@atlaskit/editor-test-helpers",
	"version": "8.0.2"
}
