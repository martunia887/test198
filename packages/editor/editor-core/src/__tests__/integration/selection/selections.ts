import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { editable, getDocFromElement, fullpage } from '../_helpers';

import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';
import blockExtensionAdf from './__fixtures__/block-extension-selection.adf.json';
import bodiedExtensionAdf from './__fixtures__/bodied-extension-selection.adf.json';
import smartLinkAdf from './__fixtures__/smart-link-selection.adf.json';
import blockCardAdf from './__fixtures__/block-card-selection.adf.json';
import mediaAdf from './__fixtures__/media-selection.adf.json';
import { EditorTestCardProvider } from '@atlaskit/editor-test-helpers';

const testDocuments: { [key: string]: string } = {
  blockExtension: blockExtensionAdf,
  bodiedExtension: bodiedExtensionAdf,
  smartLink: smartLinkAdf,
  blockCard: blockCardAdf,
  mediaSingle: mediaAdf,
};

[
  [
    'h1:first-of-type',
    'h6:first-of-type',
    'blockCard',
    'top to bottom',
    '.blockCardView-content-wrap',
  ],
  [
    'h6:first-of-type',
    'h1:first-of-type',
    'blockCard',
    'bottom to top',
    '.blockCardView-content-wrap',
  ],
  ['h1:first-of-type', 'h6:first-of-type', 'mediaSingle', 'top to bottom'],
  ['h6:first-of-type', 'h1:first-of-type', 'mediaSingle', 'bottom to top'],
  // TODO fix the dummy content provider that breaks this test.
  // ['p:first-child', 'p:last-child', 'blockExtension', 'top to bottom'],
  [
    'h6:first-of-type',
    'h1:first-of-type',
    'blockExtension',
    'bottom to top',
    '.extensionView-content-wrap',
  ],
  [
    'h1:first-of-type',
    'h6:first-of-type',
    'bodiedExtension',
    'top to bottom',
    '.bodiedExtensionView-content-wrap',
  ],
  [
    'h6:first-of-type',
    'h1:first-of-type',
    'bodiedExtension',
    'bottom to top',
    '.bodiedExtensionView-content-wrap',
  ],
  [
    'h1:first-of-type',
    'h6:first-of-type',
    'smartLink',
    'top to bottom',
    '.inlineCardView-content-wrap',
  ],
  [
    'h6:first-of-type',
    'h1:first-of-type',
    'smartLink',
    'bottom to top',
    '.inlineCardView-content-wrap',
  ],
].forEach(([startSelector, targetSelector, name, direction, initSelection]) => {
  BrowserTestCase(
    `select-extension.ts: Select ${name} with other contents from ${direction}`,
    { skip: ['edge', 'ie', 'safari'] },
    async (client: any, testName: string) => {
      const page = await goToEditorTestingExample(client);
      const cardProvider = new EditorTestCardProvider();
      await mountEditor(page, {
        appearance: fullpage.appearance,
        allowExtension: true,
        media: {
          allowMediaSingle: true,
        },
        UNSAFE_cards: { provider: Promise.resolve(cardProvider) },
        defaultValue: testDocuments[name],
      });

      if (initSelection) {
        await page.waitForSelector(initSelection);
      }

      await page.pause(1000);

      await page.simulateUserSelection(startSelector, targetSelector);

      await page.pause(500);

      await page.keys('Backspace');

      await page.pause(500);

      const doc = await page.$eval(editable, getDocFromElement);
      expect(doc).toMatchCustomDocSnapshot(testName);
    },
  );
});
