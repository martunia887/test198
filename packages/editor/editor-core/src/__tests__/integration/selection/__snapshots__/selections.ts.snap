// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select-extension.ts: Select blockCard with other contents from bottom to top 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`select-extension.ts: Select blockCard with other contents from top to bottom 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`select-extension.ts: Select blockExtension with other contents from bottom to top 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`select-extension.ts: Select bodiedExtension with other contents from bottom to top 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`select-extension.ts: Select bodiedExtension with other contents from top to bottom 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`select-extension.ts: Select mediaSingle with other contents from bottom to top 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`select-extension.ts: Select mediaSingle with other contents from top to bottom 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`select-extension.ts: Select smartLink with other contents from bottom to top 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`select-extension.ts: Select smartLink with other contents from top to bottom 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
